milla/src/lib.rs

```diff
@@ -0,0 +1,2224 @@
+    tile_atmos[ATMOS_THERMAL_ENERGY] /= heat_capacity(&tile_atmos);
+    if tile_atmos[ATMOS_THERMAL_ENERGY].is_nan() {
+        tile_atmos[ATMOS_THERMAL_ENERGY] = 0.0;
+    }
+    Ok(tile_atmos
+        .iter()
+        .map(|v| ByondValue::from(*v))
+        .collect::<Vec<ByondValue>>()
+        .as_slice()
+        .try_into()?)
+}
+
+// Rust version of fetching the atmos details of a tile.
+fn internal_get_tile_atmos(
+    x: i32,
+    y: i32,
+    z: i32,
+) -> Result<[f32; ATMOS_DEPTH], TryFromSliceError> {
+    let mut buffer = get_active_atmos_buffer_map().lock().unwrap();
+    get_atmos!(atmos, buffer, z);
+    Ok(atmos[calc_index(x, y, 0)..calc_index(x, y, ATMOS_DEPTH)].try_into()?)
+}
+
+// A tile that we consider interesting for some reason.
+struct InterestingTile {
+    x: i32,
+    y: i32,
+    z: i32,
+    // These represent the net force generated by air flowing out of the tile along each axis.
+    // For a tile losing air equally in all directions, they will be zero.
+    // For a tile losing air towards positive X only, flow_x will be the air pressure moved.
+    flow_x: f32,
+    flow_y: f32,
+    // How much "fuel" was burnt, across all reaction types, in the prior tick.
+    fuel_burnt: f32,
+}
+
+// Converts an InterestingTile into a short BYOND list.
+// (It'd be a datum, but we don't have access to those.)
+impl TryFrom<InterestingTile> for ByondValue {
+    type Error = Error;
+
+    fn try_from(value: InterestingTile) -> Result<Self, Self::Error> {
+        let result = ByondValue::new_list()?;
+        result.write_list(&[
+            // +1 here to convert from our 0-indexing to BYOND's 1-indexing.
+            ByondValue::from((value.x + 1) as f32),
+            ByondValue::from((value.y + 1) as f32),
+            ByondValue::from(value.z as f32),
+            ByondValue::from(value.flow_x),
+            ByondValue::from(value.flow_y),
+            ByondValue::from(value.fuel_burnt),
+        ])?;
+        Ok(result)
+    }
+}
+
+// BYOND API for getting a list of interesting turfs this tick.
+// There are three kinds:
+// * Turfs that are hot eough to cause fires.
+// * Turfs that just passed the threshold for showing plasma or sleeping gas.
+// * Turfs with strong airflow out.
+#[byondapi::bind]
+fn get_interesting_tiles() {
+    setup_panic_handler();
+    let mut interesting_tiles = get_interesting_tiles_vector().lock().unwrap();
+    let initial_interesting_tiles = interesting_tiles.len();
+    let byond_interesting_tiles = interesting_tiles
+        .drain(..)
+        .map(|v| ByondValue::try_from(v))
+        .flatten()
+        .collect::<Vec<ByondValue>>();
+    if byond_interesting_tiles.len() != initial_interesting_tiles {
+        return Err(eyre!(
+            "Unable to convert all values, is BYOND out of memory?"
+        ));
+    }
+    Ok(byond_interesting_tiles.as_slice().try_into()?)
+}
+
+// Calculates the pressure flow out from this tile to the two neighboring tiles on this axis.
```

Comment:
If you switch to using /// for documenting functions, then you could probably `cargo doc` this whole package.

Reply:
I tried, but it didn't work. The BYOND functions lose the comment, presumably due to `#[byondapi::bind]`, and the other functions aren't (and have no reason to be) public, so they don't generate docs at all.